File: src/main/vrml/net/coderextreme/personal/rubikFurnace.mjs
let browser = X3D.getBrowser();
let X3D0 = {};
X3D0.profile = "Immersive";
X3D0.version = "3.3";
let head1 = browser.currentScene.createNode("head");
let meta2 = browser.currentScene.createNode("meta");
meta2.name = "title";
meta2.content = "rubikFurnace.x3d";
head1.meta = new MFNode();

head1.meta[0] = meta2;

let meta3 = browser.currentScene.createNode("meta");
meta3.name = "creator";
meta3.content = "John Carlson";
head1.meta[1] = meta3;

let meta4 = browser.currentScene.createNode("meta");
meta4.name = "generator";
meta4.content = "manual";
head1.meta[2] = meta4;

let meta5 = browser.currentScene.createNode("meta");
meta5.name = "identifier";
meta5.content = "https://coderextreme.net/X3DJSONLD/src/main/data/rubikFurnace.x3d";
head1.meta[3] = meta5;

let meta6 = browser.currentScene.createNode("meta");
meta6.name = "description";
meta6.content = "a green rubik cube";
head1.meta[4] = meta6;

head = head1;

let NavigationInfo8 = browser.currentScene.createNode("NavigationInfo");
NavigationInfo8.type = new MFString(new java.lang.String["EXAMINE"]);
browser.currentScene.children = new MFNode();

browser.currentScene.children[0] = NavigationInfo8;

let Viewpoint9 = browser.currentScene.createNode("Viewpoint");
Viewpoint9.description = "Rubiks Cube on Fire";
Viewpoint9.position = new SFVec3f(new float[0,0,12]);
browser.currentScene.children[1] = Viewpoint9;

let ProtoDeclare10 = browser.createX3DFromString(`<?xml version="1.0" encoding="undefined"?>
<!DOCTYPE X3D PUBLIC "ISO//Web3D//DTD X3D undefined//EN" "https://www.web3d.org/specifications/x3d-undefined.dtd">
<ProtoDeclare name="anyShape" ><ProtoInterface><field name="xtranslation" accessType="inputOutput" type="SFVec3f" value="0 0 0"></field>
<field name="myShape" accessType="inputOutput" type="MFNode"><Shape><Sphere></Sphere>
<Appearance containerField="appearance"><Material containerField="material" diffuseColor="1 1 1"></Material>
</Appearance>
</Shape>
</field>
</ProtoInterface>
<ProtoBody><Transform><IS><connect nodeField="translation" protoField="xtranslation"></connect>
<connect nodeField="children" protoField="myShape"></connect>
</IS>
</Transform>
</ProtoBody>
</ProtoDeclare>`);
ProtoDeclare10.name = "anyShape";
let ProtoInterface11 = browser.currentScene.createNode("ProtoInterface");
let field12 = browser.currentScene.createNode("field");
field12.name = "xtranslation";
field12.accessType = "inputOutput";
field12.type = "SFVec3f";
field12.value = "0 0 0";
ProtoInterface11.field = new MFNode();

ProtoInterface11.field[0] = field12;

let field13 = browser.currentScene.createNode("field");
field13.name = "myShape";
field13.accessType = "inputOutput";
field13.type = "MFNode";
let Shape14 = browser.currentScene.createNode("Shape");
let Sphere15 = browser.currentScene.createNode("Sphere");
Shape14.geometry = Sphere15;

let Appearance16 = browser.currentScene.createNode("Appearance");
let Material17 = browser.currentScene.createNode("Material");
Material17.diffuseColor = new SFColor(new float[1,1,1]);
Appearance16.material = Material17;

Shape14.appearance = Appearance16;

field13.children = new MFNode();

field13.children[0] = Shape14;

ProtoInterface11.field[1] = field13;

ProtoDeclare10.protoInterface = ProtoInterface11;

let ProtoBody18 = browser.currentScene.createNode("ProtoBody");
let Transform19 = browser.currentScene.createNode("Transform");
let IS20 = browser.currentScene.createNode("IS");
let connect21 = browser.currentScene.createNode("connect");
connect21.nodeField = "translation";
connect21.protoField = "xtranslation";
IS20.connect = new MFNode();

IS20.connect[0] = connect21;

let connect22 = browser.currentScene.createNode("connect");
connect22.nodeField = "children";
connect22.protoField = "myShape";
IS20.connect[1] = connect22;

Transform19.iS = IS20;

ProtoBody18.children = new MFNode();

ProtoBody18.children[0] = Transform19;

ProtoDeclare10.protoBody = ProtoBody18;

browser.currentScene.children[2] = ProtoDeclare10;

let ProtoDeclare23 = browser.createX3DFromString(`<?xml version="1.0" encoding="undefined"?>
<!DOCTYPE X3D PUBLIC "ISO//Web3D//DTD X3D undefined//EN" "https://www.web3d.org/specifications/x3d-undefined.dtd">
<ProtoDeclare name="three" ><ProtoInterface><field name="ytranslation" accessType="inputOutput" type="SFVec3f" value="0 0 0"></field>
<field name="myShape" accessType="inputOutput" type="MFNode"><Shape><Sphere></Sphere>
<Appearance containerField="appearance"><Material containerField="material" diffuseColor="1 1 1"></Material>
</Appearance>
</Shape>
</field>
</ProtoInterface>
<ProtoBody><Transform><IS><connect nodeField="translation" protoField="ytranslation"></connect>
</IS>
<ProtoInstance name="anyShape"><fieldValue name="xtranslation" value="0 0 0"></fieldValue>
<IS><connect nodeField="myShape" protoField="myShape"></connect>
</IS>
</ProtoInstance>
<ProtoInstance name="anyShape"><fieldValue name="xtranslation" value="2 0 0"></fieldValue>
<IS><connect nodeField="myShape" protoField="myShape"></connect>
</IS>
</ProtoInstance>
<ProtoInstance name="anyShape"><fieldValue name="xtranslation" value="-2 0 0"></fieldValue>
<IS><connect nodeField="myShape" protoField="myShape"></connect>
</IS>
</ProtoInstance>
</Transform>
</ProtoBody>
</ProtoDeclare>`);
ProtoDeclare23.name = "three";
let ProtoInterface24 = browser.currentScene.createNode("ProtoInterface");
let field25 = browser.currentScene.createNode("field");
field25.name = "ytranslation";
field25.accessType = "inputOutput";
field25.type = "SFVec3f";
field25.value = "0 0 0";
ProtoInterface24.field = new MFNode();

ProtoInterface24.field[0] = field25;

let field26 = browser.currentScene.createNode("field");
field26.name = "myShape";
field26.accessType = "inputOutput";
field26.type = "MFNode";
let Shape27 = browser.currentScene.createNode("Shape");
let Sphere28 = browser.currentScene.createNode("Sphere");
Shape27.geometry = Sphere28;

let Appearance29 = browser.currentScene.createNode("Appearance");
let Material30 = browser.currentScene.createNode("Material");
Material30.diffuseColor = new SFColor(new float[1,1,1]);
Appearance29.material = Material30;

Shape27.appearance = Appearance29;

field26.children = new MFNode();

field26.children[0] = Shape27;

ProtoInterface24.field[1] = field26;

ProtoDeclare23.protoInterface = ProtoInterface24;

let ProtoBody31 = browser.currentScene.createNode("ProtoBody");
let Transform32 = browser.currentScene.createNode("Transform");
let IS33 = browser.currentScene.createNode("IS");
let connect34 = browser.currentScene.createNode("connect");
connect34.nodeField = "translation";
connect34.protoField = "ytranslation";
IS33.connect = new MFNode();

IS33.connect[0] = connect34;

Transform32.iS = IS33;

let ProtoInstance35 = browser.currentScene.createNode("ProtoInstance");
ProtoInstance35.name = "anyShape";
let fieldValue36 = browser.currentScene.createNode("fieldValue");
fieldValue36.name = "xtranslation";
fieldValue36.value = "0 0 0";
ProtoInstance35.fieldValue = new MFNode();

ProtoInstance35.fieldValue[0] = fieldValue36;

let IS37 = browser.currentScene.createNode("IS");
let connect38 = browser.currentScene.createNode("connect");
connect38.nodeField = "myShape";
connect38.protoField = "myShape";
IS37.connect = new MFNode();

IS37.connect[0] = connect38;

ProtoInstance35.iS = IS37;

Transform32.children = new MFNode();

Transform32.children[0] = ProtoInstance35;

let ProtoInstance39 = browser.currentScene.createNode("ProtoInstance");
ProtoInstance39.name = "anyShape";
let fieldValue40 = browser.currentScene.createNode("fieldValue");
fieldValue40.name = "xtranslation";
fieldValue40.value = "2 0 0";
ProtoInstance39.fieldValue = new MFNode();

ProtoInstance39.fieldValue[0] = fieldValue40;

let IS41 = browser.currentScene.createNode("IS");
let connect42 = browser.currentScene.createNode("connect");
connect42.nodeField = "myShape";
connect42.protoField = "myShape";
IS41.connect = new MFNode();

IS41.connect[0] = connect42;

ProtoInstance39.iS = IS41;

Transform32.children[1] = ProtoInstance39;

let ProtoInstance43 = browser.currentScene.createNode("ProtoInstance");
ProtoInstance43.name = "anyShape";
let fieldValue44 = browser.currentScene.createNode("fieldValue");
fieldValue44.name = "xtranslation";
fieldValue44.value = "-2 0 0";
ProtoInstance43.fieldValue = new MFNode();

ProtoInstance43.fieldValue[0] = fieldValue44;

let IS45 = browser.currentScene.createNode("IS");
let connect46 = browser.currentScene.createNode("connect");
connect46.nodeField = "myShape";
connect46.protoField = "myShape";
IS45.connect = new MFNode();

IS45.connect[0] = connect46;

ProtoInstance43.iS = IS45;

Transform32.children[2] = ProtoInstance43;

ProtoBody31.children = new MFNode();

ProtoBody31.children[0] = Transform32;

ProtoDeclare23.protoBody = ProtoBody31;

browser.currentScene.children[3] = ProtoDeclare23;

let ProtoDeclare47 = browser.createX3DFromString(`<?xml version="1.0" encoding="undefined"?>
<!DOCTYPE X3D PUBLIC "ISO//Web3D//DTD X3D undefined//EN" "https://www.web3d.org/specifications/x3d-undefined.dtd">
<ProtoDeclare name="nine" ><ProtoInterface><field name="ztranslation" accessType="inputOutput" type="SFVec3f" value="0 0 0"></field>
<field name="myShape" accessType="inputOutput" type="MFNode"><Shape><Sphere></Sphere>
<Appearance containerField="appearance"><Material containerField="material" diffuseColor="1 1 1"></Material>
</Appearance>
</Shape>
</field>
</ProtoInterface>
<ProtoBody><Transform><IS><connect nodeField="translation" protoField="ztranslation"></connect>
</IS>
<ProtoInstance name="three"><fieldValue name="ytranslation" value="0 0 0"></fieldValue>
<IS><connect nodeField="myShape" protoField="myShape"></connect>
</IS>
</ProtoInstance>
<ProtoInstance name="three"><fieldValue name="ytranslation" value="0 2 0"></fieldValue>
<IS><connect nodeField="myShape" protoField="myShape"></connect>
</IS>
</ProtoInstance>
<ProtoInstance name="three"><fieldValue name="ytranslation" value="0 -2 0"></fieldValue>
<IS><connect nodeField="myShape" protoField="myShape"></connect>
</IS>
</ProtoInstance>
</Transform>
</ProtoBody>
</ProtoDeclare>`);
ProtoDeclare47.name = "nine";
let ProtoInterface48 = browser.currentScene.createNode("ProtoInterface");
let field49 = browser.currentScene.createNode("field");
field49.name = "ztranslation";
field49.accessType = "inputOutput";
field49.type = "SFVec3f";
field49.value = "0 0 0";
ProtoInterface48.field = new MFNode();

ProtoInterface48.field[0] = field49;

let field50 = browser.currentScene.createNode("field");
field50.name = "myShape";
field50.accessType = "inputOutput";
field50.type = "MFNode";
let Shape51 = browser.currentScene.createNode("Shape");
let Sphere52 = browser.currentScene.createNode("Sphere");
Shape51.geometry = Sphere52;

let Appearance53 = browser.currentScene.createNode("Appearance");
let Material54 = browser.currentScene.createNode("Material");
Material54.diffuseColor = new SFColor(new float[1,1,1]);
Appearance53.material = Material54;

Shape51.appearance = Appearance53;

field50.children = new MFNode();

field50.children[0] = Shape51;

ProtoInterface48.field[1] = field50;

ProtoDeclare47.protoInterface = ProtoInterface48;

let ProtoBody55 = browser.currentScene.createNode("ProtoBody");
let Transform56 = browser.currentScene.createNode("Transform");
let IS57 = browser.currentScene.createNode("IS");
let connect58 = browser.currentScene.createNode("connect");
connect58.nodeField = "translation";
connect58.protoField = "ztranslation";
IS57.connect = new MFNode();

IS57.connect[0] = connect58;

Transform56.iS = IS57;

let ProtoInstance59 = browser.currentScene.createNode("ProtoInstance");
ProtoInstance59.name = "three";
let fieldValue60 = browser.currentScene.createNode("fieldValue");
fieldValue60.name = "ytranslation";
fieldValue60.value = "0 0 0";
ProtoInstance59.fieldValue = new MFNode();

ProtoInstance59.fieldValue[0] = fieldValue60;

let IS61 = browser.currentScene.createNode("IS");
let connect62 = browser.currentScene.createNode("connect");
connect62.nodeField = "myShape";
connect62.protoField = "myShape";
IS61.connect = new MFNode();

IS61.connect[0] = connect62;

ProtoInstance59.iS = IS61;

Transform56.children = new MFNode();

Transform56.children[0] = ProtoInstance59;

let ProtoInstance63 = browser.currentScene.createNode("ProtoInstance");
ProtoInstance63.name = "three";
let fieldValue64 = browser.currentScene.createNode("fieldValue");
fieldValue64.name = "ytranslation";
fieldValue64.value = "0 2 0";
ProtoInstance63.fieldValue = new MFNode();

ProtoInstance63.fieldValue[0] = fieldValue64;

let IS65 = browser.currentScene.createNode("IS");
let connect66 = browser.currentScene.createNode("connect");
connect66.nodeField = "myShape";
connect66.protoField = "myShape";
IS65.connect = new MFNode();

IS65.connect[0] = connect66;

ProtoInstance63.iS = IS65;

Transform56.children[1] = ProtoInstance63;

let ProtoInstance67 = browser.currentScene.createNode("ProtoInstance");
ProtoInstance67.name = "three";
let fieldValue68 = browser.currentScene.createNode("fieldValue");
fieldValue68.name = "ytranslation";
fieldValue68.value = "0 -2 0";
ProtoInstance67.fieldValue = new MFNode();

ProtoInstance67.fieldValue[0] = fieldValue68;

let IS69 = browser.currentScene.createNode("IS");
let connect70 = browser.currentScene.createNode("connect");
connect70.nodeField = "myShape";
connect70.protoField = "myShape";
IS69.connect = new MFNode();

IS69.connect[0] = connect70;

ProtoInstance67.iS = IS69;

Transform56.children[2] = ProtoInstance67;

ProtoBody55.children = new MFNode();

ProtoBody55.children[0] = Transform56;

ProtoDeclare47.protoBody = ProtoBody55;

browser.currentScene.children[4] = ProtoDeclare47;

let ProtoDeclare71 = browser.createX3DFromString(`<?xml version="1.0" encoding="undefined"?>
<!DOCTYPE X3D PUBLIC "ISO//Web3D//DTD X3D undefined//EN" "https://www.web3d.org/specifications/x3d-undefined.dtd">
<ProtoDeclare name="twentyseven" ><ProtoInterface><field name="ttranslation" accessType="inputOutput" type="SFVec3f" value="0 0 0"></field>
<field name="myShape" accessType="inputOutput" type="MFNode"><Shape><Sphere></Sphere>
<Appearance containerField="appearance"><Material containerField="material" diffuseColor="1 1 1"></Material>
</Appearance>
</Shape>
</field>
</ProtoInterface>
<ProtoBody><Transform><IS><connect nodeField="translation" protoField="ttranslation"></connect>
</IS>
<ProtoInstance name="nine"><fieldValue name="ztranslation" value="0 0 0"></fieldValue>
<IS><connect nodeField="myShape" protoField="myShape"></connect>
</IS>
</ProtoInstance>
<ProtoInstance name="nine"><fieldValue name="ztranslation" value="0 0 2"></fieldValue>
<IS><connect nodeField="myShape" protoField="myShape"></connect>
</IS>
</ProtoInstance>
<ProtoInstance name="nine"><fieldValue name="ztranslation" value="0 0 -2"></fieldValue>
<IS><connect nodeField="myShape" protoField="myShape"></connect>
</IS>
</ProtoInstance>
</Transform>
</ProtoBody>
</ProtoDeclare>`);
ProtoDeclare71.name = "twentyseven";
let ProtoInterface72 = browser.currentScene.createNode("ProtoInterface");
let field73 = browser.currentScene.createNode("field");
field73.name = "ttranslation";
field73.accessType = "inputOutput";
field73.type = "SFVec3f";
field73.value = "0 0 0";
ProtoInterface72.field = new MFNode();

ProtoInterface72.field[0] = field73;

let field74 = browser.currentScene.createNode("field");
field74.name = "myShape";
field74.accessType = "inputOutput";
field74.type = "MFNode";
let Shape75 = browser.currentScene.createNode("Shape");
let Sphere76 = browser.currentScene.createNode("Sphere");
Shape75.geometry = Sphere76;

let Appearance77 = browser.currentScene.createNode("Appearance");
let Material78 = browser.currentScene.createNode("Material");
Material78.diffuseColor = new SFColor(new float[1,1,1]);
Appearance77.material = Material78;

Shape75.appearance = Appearance77;

field74.children = new MFNode();

field74.children[0] = Shape75;

ProtoInterface72.field[1] = field74;

ProtoDeclare71.protoInterface = ProtoInterface72;

let ProtoBody79 = browser.currentScene.createNode("ProtoBody");
let Transform80 = browser.currentScene.createNode("Transform");
let IS81 = browser.currentScene.createNode("IS");
let connect82 = browser.currentScene.createNode("connect");
connect82.nodeField = "translation";
connect82.protoField = "ttranslation";
IS81.connect = new MFNode();

IS81.connect[0] = connect82;

Transform80.iS = IS81;

let ProtoInstance83 = browser.currentScene.createNode("ProtoInstance");
ProtoInstance83.name = "nine";
let fieldValue84 = browser.currentScene.createNode("fieldValue");
fieldValue84.name = "ztranslation";
fieldValue84.value = "0 0 0";
ProtoInstance83.fieldValue = new MFNode();

ProtoInstance83.fieldValue[0] = fieldValue84;

let IS85 = browser.currentScene.createNode("IS");
let connect86 = browser.currentScene.createNode("connect");
connect86.nodeField = "myShape";
connect86.protoField = "myShape";
IS85.connect = new MFNode();

IS85.connect[0] = connect86;

ProtoInstance83.iS = IS85;

Transform80.children = new MFNode();

Transform80.children[0] = ProtoInstance83;

let ProtoInstance87 = browser.currentScene.createNode("ProtoInstance");
ProtoInstance87.name = "nine";
let fieldValue88 = browser.currentScene.createNode("fieldValue");
fieldValue88.name = "ztranslation";
fieldValue88.value = "0 0 2";
ProtoInstance87.fieldValue = new MFNode();

ProtoInstance87.fieldValue[0] = fieldValue88;

let IS89 = browser.currentScene.createNode("IS");
let connect90 = browser.currentScene.createNode("connect");
connect90.nodeField = "myShape";
connect90.protoField = "myShape";
IS89.connect = new MFNode();

IS89.connect[0] = connect90;

ProtoInstance87.iS = IS89;

Transform80.children[1] = ProtoInstance87;

let ProtoInstance91 = browser.currentScene.createNode("ProtoInstance");
ProtoInstance91.name = "nine";
let fieldValue92 = browser.currentScene.createNode("fieldValue");
fieldValue92.name = "ztranslation";
fieldValue92.value = "0 0 -2";
ProtoInstance91.fieldValue = new MFNode();

ProtoInstance91.fieldValue[0] = fieldValue92;

let IS93 = browser.currentScene.createNode("IS");
let connect94 = browser.currentScene.createNode("connect");
connect94.nodeField = "myShape";
connect94.protoField = "myShape";
IS93.connect = new MFNode();

IS93.connect[0] = connect94;

ProtoInstance91.iS = IS93;

Transform80.children[2] = ProtoInstance91;

ProtoBody79.children = new MFNode();

ProtoBody79.children[0] = Transform80;

ProtoDeclare71.protoBody = ProtoBody79;

browser.currentScene.children[5] = ProtoDeclare71;

let ProtoInstance95 = browser.currentScene.createNode("ProtoInstance");
ProtoInstance95.name = "twentyseven";
let fieldValue96 = browser.currentScene.createNode("fieldValue");
fieldValue96.name = "ttranslation";
fieldValue96.value = "0 0 0";
ProtoInstance95.fieldValue = new MFNode();

ProtoInstance95.fieldValue[0] = fieldValue96;

let fieldValue97 = browser.currentScene.createNode("fieldValue");
fieldValue97.name = "myShape";
let Shape98 = browser.currentScene.createNode("Shape");
let Box99 = browser.currentScene.createNode("Box");
Box99.size = new SFVec3f(new float[1,1,1]);
Shape98.geometry = Box99;

let Appearance100 = browser.currentScene.createNode("Appearance");
let Material101 = browser.currentScene.createNode("Material");
Material101.diffuseColor = new SFColor(new float[0,1,0]);
Appearance100.material = Material101;

Shape98.appearance = Appearance100;

fieldValue97.children = new MFNode();

fieldValue97.children[0] = Shape98;

ProtoInstance95.fieldValue[1] = fieldValue97;

browser.currentScene.children[6] = ProtoInstance95;

File: src/main/python/net/coderextreme/personal/rubikFurnace.py
import x3dpsail as x3d
X3D0 = x3d.X3D()
X3D0.setProfile("Immersive")
X3D0.setVersion("3.3")
head1 = x3d.head()
meta2 = x3d.meta()
meta2.setName("title")
meta2.setContent("rubikFurnace.x3d")

head1.addMeta(meta2)
meta3 = x3d.meta()
meta3.setName("creator")
meta3.setContent("John Carlson")

head1.addMeta(meta3)
meta4 = x3d.meta()
meta4.setName("generator")
meta4.setContent("manual")

head1.addMeta(meta4)
meta5 = x3d.meta()
meta5.setName("identifier")
meta5.setContent("https://coderextreme.net/X3DJSONLD/src/main/personal/rubikFurnace.x3d")

head1.addMeta(meta5)
meta6 = x3d.meta()
meta6.setName("description")
meta6.setContent("a green rubik cube")

head1.addMeta(meta6)

X3D0.setHead(head1)
Scene7 = x3d.Scene()
NavigationInfo8 = x3d.NavigationInfo()
NavigationInfo8.setType(["EXAMINE"])

Scene7.addChildren(NavigationInfo8)
Viewpoint9 = x3d.Viewpoint()
Viewpoint9.setDescription("Rubiks Cube on Fire")
Viewpoint9.setPosition([0,0,12])

Scene7.addChildren(Viewpoint9)
ProtoDeclare10 = x3d.ProtoDeclare()
ProtoDeclare10.setName("anyShape")
ProtoInterface11 = x3d.ProtoInterface()
field12 = x3d.field()
field12.setName("xtranslation")
field12.setAccessType("inputOutput")
field12.setType("SFVec3f")
field12.setValue("0 0 0")

ProtoInterface11.addField(field12)
field13 = x3d.field()
field13.setName("myShape")
field13.setAccessType("inputOutput")
field13.setType("MFNode")
Shape14 = x3d.Shape()
Sphere15 = x3d.Sphere()

Shape14.setGeometry(Sphere15)
Appearance16 = x3d.Appearance()
Material17 = x3d.Material()
Material17.setDiffuseColor([1,1,1])

Appearance16.setMaterial(Material17)

Shape14.setAppearance(Appearance16)

field13.addChildren(Shape14)

ProtoInterface11.addField(field13)

ProtoDeclare10.setProtoInterface(ProtoInterface11)
ProtoBody18 = x3d.ProtoBody()
Transform19 = x3d.Transform()
IS20 = x3d.IS()
connect21 = x3d.connect()
connect21.setNodeField("translation")
connect21.setProtoField("xtranslation")

IS20.addConnect(connect21)
connect22 = x3d.connect()
connect22.setNodeField("children")
connect22.setProtoField("myShape")

IS20.addConnect(connect22)

Transform19.setIS(IS20)

ProtoBody18.addChildren(Transform19)

ProtoDeclare10.setProtoBody(ProtoBody18)

Scene7.addChildren(ProtoDeclare10)
ProtoDeclare23 = x3d.ProtoDeclare()
ProtoDeclare23.setName("three")
ProtoInterface24 = x3d.ProtoInterface()
field25 = x3d.field()
field25.setName("ytranslation")
field25.setAccessType("inputOutput")
field25.setType("SFVec3f")
field25.setValue("0 0 0")

ProtoInterface24.addField(field25)
field26 = x3d.field()
field26.setName("myShape")
field26.setAccessType("inputOutput")
field26.setType("MFNode")
Shape27 = x3d.Shape()
Sphere28 = x3d.Sphere()

Shape27.setGeometry(Sphere28)
Appearance29 = x3d.Appearance()
Material30 = x3d.Material()
Material30.setDiffuseColor([1,1,1])

Appearance29.setMaterial(Material30)

Shape27.setAppearance(Appearance29)

field26.addChildren(Shape27)

ProtoInterface24.addField(field26)

ProtoDeclare23.setProtoInterface(ProtoInterface24)
ProtoBody31 = x3d.ProtoBody()
Transform32 = x3d.Transform()
IS33 = x3d.IS()
connect34 = x3d.connect()
connect34.setNodeField("translation")
connect34.setProtoField("ytranslation")

IS33.addConnect(connect34)

Transform32.setIS(IS33)
ProtoInstance35 = x3d.ProtoInstance()
ProtoInstance35.setName("anyShape")
fieldValue36 = x3d.fieldValue()
fieldValue36.setName("xtranslation")
fieldValue36.setValue("0 0 0")

ProtoInstance35.addFieldValue(fieldValue36)
IS37 = x3d.IS()
connect38 = x3d.connect()
connect38.setNodeField("myShape")
connect38.setProtoField("myShape")

IS37.addConnect(connect38)

ProtoInstance35.setIS(IS37)

Transform32.addChildren(ProtoInstance35)
ProtoInstance39 = x3d.ProtoInstance()
ProtoInstance39.setName("anyShape")
fieldValue40 = x3d.fieldValue()
fieldValue40.setName("xtranslation")
fieldValue40.setValue("2 0 0")

ProtoInstance39.addFieldValue(fieldValue40)
IS41 = x3d.IS()
connect42 = x3d.connect()
connect42.setNodeField("myShape")
connect42.setProtoField("myShape")

IS41.addConnect(connect42)

ProtoInstance39.setIS(IS41)

Transform32.addChildren(ProtoInstance39)
ProtoInstance43 = x3d.ProtoInstance()
ProtoInstance43.setName("anyShape")
fieldValue44 = x3d.fieldValue()
fieldValue44.setName("xtranslation")
fieldValue44.setValue("-2 0 0")

ProtoInstance43.addFieldValue(fieldValue44)
IS45 = x3d.IS()
connect46 = x3d.connect()
connect46.setNodeField("myShape")
connect46.setProtoField("myShape")

IS45.addConnect(connect46)

ProtoInstance43.setIS(IS45)

Transform32.addChildren(ProtoInstance43)

ProtoBody31.addChildren(Transform32)

ProtoDeclare23.setProtoBody(ProtoBody31)

Scene7.addChildren(ProtoDeclare23)
ProtoDeclare47 = x3d.ProtoDeclare()
ProtoDeclare47.setName("nine")
ProtoInterface48 = x3d.ProtoInterface()
field49 = x3d.field()
field49.setName("ztranslation")
field49.setAccessType("inputOutput")
field49.setType("SFVec3f")
field49.setValue("0 0 0")

ProtoInterface48.addField(field49)
field50 = x3d.field()
field50.setName("myShape")
field50.setAccessType("inputOutput")
field50.setType("MFNode")
Shape51 = x3d.Shape()
Sphere52 = x3d.Sphere()

Shape51.setGeometry(Sphere52)
Appearance53 = x3d.Appearance()
Material54 = x3d.Material()
Material54.setDiffuseColor([1,1,1])

Appearance53.setMaterial(Material54)

Shape51.setAppearance(Appearance53)

field50.addChildren(Shape51)

ProtoInterface48.addField(field50)

ProtoDeclare47.setProtoInterface(ProtoInterface48)
ProtoBody55 = x3d.ProtoBody()
Transform56 = x3d.Transform()
IS57 = x3d.IS()
connect58 = x3d.connect()
connect58.setNodeField("translation")
connect58.setProtoField("ztranslation")

IS57.addConnect(connect58)

Transform56.setIS(IS57)
ProtoInstance59 = x3d.ProtoInstance()
ProtoInstance59.setName("three")
fieldValue60 = x3d.fieldValue()
fieldValue60.setName("ytranslation")
fieldValue60.setValue("0 0 0")

ProtoInstance59.addFieldValue(fieldValue60)
IS61 = x3d.IS()
connect62 = x3d.connect()
connect62.setNodeField("myShape")
connect62.setProtoField("myShape")

IS61.addConnect(connect62)

ProtoInstance59.setIS(IS61)

Transform56.addChildren(ProtoInstance59)
ProtoInstance63 = x3d.ProtoInstance()
ProtoInstance63.setName("three")
fieldValue64 = x3d.fieldValue()
fieldValue64.setName("ytranslation")
fieldValue64.setValue("0 2 0")

ProtoInstance63.addFieldValue(fieldValue64)
IS65 = x3d.IS()
connect66 = x3d.connect()
connect66.setNodeField("myShape")
connect66.setProtoField("myShape")

IS65.addConnect(connect66)

ProtoInstance63.setIS(IS65)

Transform56.addChildren(ProtoInstance63)
ProtoInstance67 = x3d.ProtoInstance()
ProtoInstance67.setName("three")
fieldValue68 = x3d.fieldValue()
fieldValue68.setName("ytranslation")
fieldValue68.setValue("0 -2 0")

ProtoInstance67.addFieldValue(fieldValue68)
IS69 = x3d.IS()
connect70 = x3d.connect()
connect70.setNodeField("myShape")
connect70.setProtoField("myShape")

IS69.addConnect(connect70)

ProtoInstance67.setIS(IS69)

Transform56.addChildren(ProtoInstance67)

ProtoBody55.addChildren(Transform56)

ProtoDeclare47.setProtoBody(ProtoBody55)

Scene7.addChildren(ProtoDeclare47)
ProtoDeclare71 = x3d.ProtoDeclare()
ProtoDeclare71.setName("twentyseven")
ProtoInterface72 = x3d.ProtoInterface()
field73 = x3d.field()
field73.setName("ttranslation")
field73.setAccessType("inputOutput")
field73.setType("SFVec3f")
field73.setValue("0 0 0")

ProtoInterface72.addField(field73)
field74 = x3d.field()
field74.setName("myShape")
field74.setAccessType("inputOutput")
field74.setType("MFNode")
Shape75 = x3d.Shape()
Sphere76 = x3d.Sphere()

Shape75.setGeometry(Sphere76)
Appearance77 = x3d.Appearance()
Material78 = x3d.Material()
Material78.setDiffuseColor([1,1,1])

Appearance77.setMaterial(Material78)

Shape75.setAppearance(Appearance77)

field74.addChildren(Shape75)

ProtoInterface72.addField(field74)

ProtoDeclare71.setProtoInterface(ProtoInterface72)
ProtoBody79 = x3d.ProtoBody()
Transform80 = x3d.Transform()
IS81 = x3d.IS()
connect82 = x3d.connect()
connect82.setNodeField("translation")
connect82.setProtoField("ttranslation")

IS81.addConnect(connect82)

Transform80.setIS(IS81)
ProtoInstance83 = x3d.ProtoInstance()
ProtoInstance83.setName("nine")
fieldValue84 = x3d.fieldValue()
fieldValue84.setName("ztranslation")
fieldValue84.setValue("0 0 0")

ProtoInstance83.addFieldValue(fieldValue84)
IS85 = x3d.IS()
connect86 = x3d.connect()
connect86.setNodeField("myShape")
connect86.setProtoField("myShape")

IS85.addConnect(connect86)

ProtoInstance83.setIS(IS85)

Transform80.addChildren(ProtoInstance83)
ProtoInstance87 = x3d.ProtoInstance()
ProtoInstance87.setName("nine")
fieldValue88 = x3d.fieldValue()
fieldValue88.setName("ztranslation")
fieldValue88.setValue("0 0 2")

ProtoInstance87.addFieldValue(fieldValue88)
IS89 = x3d.IS()
connect90 = x3d.connect()
connect90.setNodeField("myShape")
connect90.setProtoField("myShape")

IS89.addConnect(connect90)

ProtoInstance87.setIS(IS89)

Transform80.addChildren(ProtoInstance87)
ProtoInstance91 = x3d.ProtoInstance()
ProtoInstance91.setName("nine")
fieldValue92 = x3d.fieldValue()
fieldValue92.setName("ztranslation")
fieldValue92.setValue("0 0 -2")

ProtoInstance91.addFieldValue(fieldValue92)
IS93 = x3d.IS()
connect94 = x3d.connect()
connect94.setNodeField("myShape")
connect94.setProtoField("myShape")

IS93.addConnect(connect94)

ProtoInstance91.setIS(IS93)

Transform80.addChildren(ProtoInstance91)

ProtoBody79.addChildren(Transform80)

ProtoDeclare71.setProtoBody(ProtoBody79)

Scene7.addChildren(ProtoDeclare71)
ProtoInstance95 = x3d.ProtoInstance()
ProtoInstance95.setName("twentyseven")
fieldValue96 = x3d.fieldValue()
fieldValue96.setName("ttranslation")
fieldValue96.setValue("0 0 0")

ProtoInstance95.addFieldValue(fieldValue96)
fieldValue97 = x3d.fieldValue()
fieldValue97.setName("myShape")
Shape98 = x3d.Shape()
Box99 = x3d.Box()
Box99.setSize([1,1,1])

Shape98.setGeometry(Box99)
Appearance100 = x3d.Appearance()
Material101 = x3d.Material()
Material101.setDiffuseColor([0,1,0])

Appearance100.setMaterial(Material101)

Shape98.setAppearance(Appearance100)

fieldValue97.addChildren(Shape98)

ProtoInstance95.addFieldValue(fieldValue97)

Scene7.addChildren(ProtoInstance95)

X3D0.setScene(Scene7)
X3D0.toFileX3D("../personal/rubikFurnace.new.python.x3d")
X3D0.toFileJSON("../personal/rubikFurnace.new.python.json")
